packages/server-side-render/src/server-side-render.js

```diff
@@ -83,17 +83,17 @@ export class ServerSideRender extends Component {
 
 	render() {
 		const response = this.state.response;
-		const { className } = this.props;
+		const { className, onEmptyResponse, onErrorResponse, onNullResponse } = this.props;
 		if ( response === '' ) {
-			return (
+			return !! onEmptyResponse ? onEmptyResponse( this ) : (
 				<Placeholder
 					className={ className }
 				>
 					{ __( 'Block rendered as empty.' ) }
```

Comment:
I wonder if onEmptyResponse should be rendered inside the placeholder? The advantage is that it is easier to use, one does not need to pass the placeholder. The disadvantage is that it is not as customizable.

Reply:
I did it this way so we can pass a custom classname to Placeholder, or allow flexibility to use something completely different.